File: src/monitor-orchestrator/utils/src/main/java/org/unict/ing/iot/utils/model/Tank.java
/* 
 * Copyright (C) 2018 aleskandro - eMarco - cursedLondor
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package org.unict.ing.iot.utils.model;

import java.io.Serializable;

/**
 *
 * @author aleskandro
 */
public final class Tank extends GenericValue implements Serializable {
    private float capacity;
    private float inputFlowRate;
    private float outputFlowRate;
    private int   tankId;
    
    private final SchmidtTrigger trigger;
    private final Electrovalve   valve;

    public Tank(float capacity, float inputFlowRate, float outputFlowRate, int tankId, SchmidtTrigger trigger, Electrovalve valve) {
        this.capacity       = capacity;
        this.inputFlowRate  = inputFlowRate;
        this.outputFlowRate = outputFlowRate;
        this.trigger        = trigger;
        this.valve          = valve;
        this.tankId         = tankId;
    }

    public Tank(float capacity, float inputFlowRate, float outputFlowRate, int tankId) {
        this(capacity, inputFlowRate, outputFlowRate, tankId, new SchmidtTrigger(), new Electrovalve());
    }

    public Tank() {
        this(0, 0, 0, 0);
    }

    public float getCapacity() {
        return capacity;
    }

    public void setCapacity(float capacity) {
        this.capacity = capacity;
    }

    public float getInputFlowRate() {
        return inputFlowRate;
    }

    public void setInputFlowRate(float inputFlowRate) {
        this.inputFlowRate = inputFlowRate;
    }

    public float getOutputFlowRate() {
        return outputFlowRate;
    }

    public void setOutputFlowRate(float outputFlowRate) {
        this.outputFlowRate = outputFlowRate;
    }

    public SchmidtTrigger getTrigger() {
        return trigger;
    }

    public Electrovalve getValve() {
        return valve;
    }

    public int getTankId() {
        return tankId;
    }

    public void setTankId(int tankId) {
        this.tankId = tankId;
    }

}

File: src/web-frontend/src/app/model/tank.ts
import { GenericValue } from './generic-value';
import { Trigger } from './trigger';
import { Valve } from './valve';

export class Tank extends GenericValue {
  capacity: number;

  inputFlowRate: number;
  outputFlowRate: number;

  tankId: number;

  trigger: Trigger;

  valve: Valve;


  static labels: string[] = ["Timestamp", "Capacity", "Input Flow Rate", "Output Flow Rate", "ID"];
  static y_label: string = "MB";
  static toArray(stat : Tank): any {
    return [new Date(stat.id.timestamp*1000), stat.capacity, stat.inputFlowRate, stat.outputFlowRate, stat.tankId];
  }
}
/*
{"className":"org.unict.ing.iot.utils.model.Tank",
"capacity":372.7488,
"inputFlowRate":3.0681,
"outputFlowRate":3.7275E-7,
"tankId":1,
"trigger":{"opened":true},
"valve":{"flowRateResistance":100.0},
"id":{"timestamp":1519597583,"machineIdentifier":2799516,"processIdentifier":64,"counter":12170025,"date":1519597583000,"time":1519597583000,"timeSecond":1519597583}
}
*/
